src/display.rs

```diff
@@ -495,8 +495,10 @@ impl Display {
                     cell_height: ch,
                     padding_x: px,
                     padding_y: py, ..} = *terminal.size_info();
-        let nspot_y = (py + (row + 1) as f32 * ch) as i32;
-        let nspot_x = (px + col as f32 * cw) as i32;
+        let dpr = self.window().hidpi_factor() as f32;
+
+        let nspot_y = (py + (row + 1) as f32 * ch / dpr) as i32;
+        let nspot_x = (px + col as f32 * cw / dpr) as i32;
```

Comment:
You're not including the padding in the DPI calculation here, is that intentional? We scale our padding by DPI, so I think it might be relevant to include it? I don't have any IME thingy setup myself, but it should be simple to test this on a monitor without 1.0 DPI by setting the padding to a large value, that should make things obvious.

If this is indeed the case, I'd propose going for a different approach though. To make sure things are done properly I'd construct a `PhysicalPosition` using `nspot_y` and `nspot_x` completely unscaled, then applying the scaling when calling `psize.to_logical(dpr)`. That should make sure the calculation is done properly.